parsel/unified.py

```diff
@@ -66,26 +56,28 @@ class Selector(object_ref):
     }
     _lxml_smart_strings = False
 
-    def __init__(self, response=None, text=None, type=None, namespaces=None,
-                 _root=None, _expr=None):
-        self.type = st = _st(response, type or self._default_type)
+    def __init__(self, text=None, type=None, namespaces=None, _root=None, _expr=None):
```

Comment:
what about renaming `root` argument - we're exposing it to user in `"Selector needs either text or _root argument"` exception message, so it is no longer private